### DISCUSSION TITLE
Translations in Wavelog

### DISCUSSION THREAD

**Discussion by HB9HIL** [wavelog/wavelog#491] [Announcements]
Translating Wavelog is a big task and needs continuous contributions from the community. We already have 15 additional languages (July 2024) that are partially translated in addition to our base language, English. To improve the translations and add more languages in the future, we need the help of all users and the international hamradio community.

To make translating Wavelog as easy as possible, we stopped working with the CodeIgniter built-in language engine and switched over to gettext() and po files, which brings a huge benefit to our development workflow.

In addition to that, we created a platform based on the open-source software Weblate to help contributors who aren't skilled with raw code. In this wiki, we want to explain how you can translate and make Wavelog accessible to as many nations as possible.

## Help us with Translations!

The best way to translate Wavelog is to use our newly created translation-platform based on Weblate. Register and you can directly start with contributions to our multilanguage-system. It's super simple and helps to make Wavelog available to as many nations as possible. Thank you for your help!

[![Übersetzungsstatus](https://translate.wavelog.org/widget/wavelog/main-translation/open-graph.png)](https://translate.wavelog.org/engage/wavelog/)

**Comment by HB9HIL** [comment]
There is now the language for the Wavelog Installer available!

https://translate.wavelog.org/projects/wavelog

**Comment by HB9HIL** [comment]
@kikosgc is currently doing a huge job in translating the newly added language portuguese! Thank you very much for contributing!

**Reply by kikosgc** [reply]
The Portuguese (pt_PT) 🇵🇹 translations are now finished! 🎉🎉🎉

**Reply by HB9HIL** [reply]
Amazing!! During development of Wavelog there will be more and more strings to translate. You may get an notifications sometimes to translate new strings :D 

Absolutely fantastic job!

**Comment by HB9HIL** [comment]
@filipmelik You may want to support the czech translation? 😬 -> [https://translate.wavelog.org/engage/wavelog/-/cs/](https://translate.wavelog.org/engage/wavelog/-/cs/)

**Reply by filipmelik** [reply]
Well, not promising anything, since there is not much time to spare, but I might try to tackle at least few of them. I will ask other OK hamradio I know if they will not be willing to help too, but as I said, not promising anything 😬

**Reply by HB9HIL** [reply]
No problem :) just through I could give a hint here. 

> [!TIP]
> There is a OpenAI auto suggestion implemented which makes it really efficient. You don't have to type everything, just crossread, correct and save.

**Comment by PE1PQX** [comment]
I have made a few Dutch translations.. Hope is helps!

**Reply by HB9HIL** [reply]
Awesome!! We need always translations. To make it easier for you you can use the automatic suggestions generated by OpenAI and just check spelling and sense. Thank you!

**Reply by PE1PQX** [reply]
Thanks.. Will have a look, time permitting... (busy here also ;) )

**Reply by pa8s** [reply]
As a fellow Dutchman, I took the liberty of translating some text via Weblate as well. I use the English version myself but maybe I can use Wavelog in our club when the club station features are released. After all, not everyone has the same proficiency in English, especially some older members.

**Reply by HB9HIL** [reply]
@pa8s  merged in #1553 :) thank you very much!!

**Reply by PE1PQX** [reply]
> As a fellow Dutchman, I took the liberty of translating some text via Weblate as well. I use the English version myself but maybe I can use Wavelog in our club when the club station features are released. After all, not everyone has the same proficiency in English, especially some older members.

Not a problem.

**Reply by PE1PQX** [reply]
PA8S and I together made translations into Dutch. PA8S did the majority of the task.
As far as I can see: Dutch translations are done!

**Reply by HB9HIL** [reply]
@PE1PQX This is awesome!!! Thank you so much! Due to the continuous development there will be new strings from time to time :-)

**Reply by pa8s** [reply]
> Due to the continuous development there will be new strings from time to time :-)

I saw some new untranslated texts this afternoon, so I immediately translated them into Dutch. From time to time I will keep an eye on the translate site for new texts to be translated.

**Reply by PE1PQX** [reply]
Do not forget to thank @pa8s also!

**Reply by HB9HIL** [reply]
@PE1PQX @pa8s 

I already added both of you guys :-) https://github.com/wavelog/wavelog/tree/dev?tab=readme-ov-file#contributing

**Comment by MaviKulubeliAdam** [comment]
The Turkish translation of Wavelog is currently 51% complete and work is ongoing.
Could you let us know when the current translations will be deployed to the live version?

73,
TA4AQG - SP9AQG Erkin

**Reply by phl0** [reply]
> Could you let us know when the current translations will be deployed to the live version?

We merge translations to the master branch upon the next release as the latest. Merges to dev branch happen from time to time. Currently no release is planned but there will be one sooner or later. 

What do you mean with "live version"?

**Reply by MaviKulubeliAdam** [reply]
I meant the merge to the master branch. Got it, thanks! Really looking forward to the next update! 😄

**Comment by OM0AAO** [comment]
Is possible to test a translation before goes public? There are some phrases which are not easy to understand meaning (context),

**Reply by AndreasK79** [reply]
It is possible, but you need to switch to the dev branch. That branch will contain the new translations (once they are merged).

**Reply by HB9HIL** [reply]
As @AndreasK79 said. The other thing is you can see where the questionable string is coming from by clicking the source link
<img width="2189" height="1026" alt="grafik" src="https://github.com/user-attachments/assets/9e92528e-7305-4e63-9646-5b422b3a8f13" />
<img width="1854" height="809" alt="grafik" src="https://github.com/user-attachments/assets/82ff9706-6b07-48eb-baee-0fe85c9b0546" />


Then you atleast have a little bit of context.

**Comment by OM0AAO** [comment]
Slovak translation finished. Weblate is absolute fantastic. Great UX/UI!

There could be errors. Not sure about some capital letters in phrases (in Slovak language are capital letters used not the same as is English). I do my best but sure it is not without error. I will try to improve it later when I see it on WaveLog.

**Reply by HB9HIL** [reply]
Thanks for your contribution!

**Comment by R1BLH** [comment]
@HB9HIL have you seen the issue with the Weblate site?

<img width="1431" height="882" alt="изображение" src="https://github.com/user-attachments/assets/065f17ce-4a5a-45d0-8a2b-d70dcd0fbdcf" />

**Reply by HB9HIL** [reply]
thx for the hint. I know whats the problem. Gonna fix it
